core/src/main/java/org/apache/calcite/rel/rules/ProjectJoinRemoveRule.java

```diff
@@ -127,6 +151,49 @@ public ProjectJoinRemoveRule(
     call.transformTo(node);
   }
 
+  /**
+   * Check as following:
+   * 1. Make sure that every foreign column always be foreign key.
+   * 2. The target of foreign key is the correct unique key correspondingly.
```

Comment:
Same as per the other "linguistic" comments:

```suggestion
   * 2. The target of foreign key is the correct corresponding unique key.
```